micrometer-observation/src/main/java/io/micrometer/observation/SimpleObservation.java

```diff
@@ -46,7 +47,7 @@ class SimpleObservation implements Observation {
 
     private final Collection<ObservationFilter> filters;
 
-    SimpleObservation(String name, ObservationRegistry registry, Context context) {
+    SimpleObservation(@Nullable String name, ObservationRegistry registry, Context context) {
```

Comment:
Are you sure that this can be null here?

Reply:
Yeah I mean look at line 60, we literally `null` the name

Reply:
I get that a `DocumentedObservation`'s `getName()` can return `null` for filtering purposes but if I take a step back, that might be valid for the `DocumentedObservation` but I think `SimpleObservation` should not accept `null` here. What should the meter handler or the tracing handler do in that case?

Also, `DocumentedObservation` has a null-check so that nullability behavior can be isolated there I guess.